### PR TITLE
Update electron and Clojurescript

### PR DESCRIPTION
**Pull request by prertik**

### CODE REVIEW

File: deploy/core/node_modules/lighttable/browserInjection.js [hunk 1]
```diff
@@ -101,7 +101,7 @@
         meta["no-inspect"] = true;
         var final = safeStringify(exp);
       }
-      ipc.sendToHost("browser-raise", [meta.obj, meta.ev, {result: final, meta: meta}]);
+      ipcRenderer.sendToHost("browser-raise", [meta.obj, meta.ev, {result: final, meta: meta}]);
```
[the remark this answers is not in the document]
Comment (Mouvedia) [reply deploy/core/node_modules/lighttable/browserInjection.js]: It's really minor.
ref http://linterrors.com/js/a-used-out-of-scope

```js
      var final;

      if (meta.ev === "editor.eval.cljs.watch")
        final = cljs.core.pr_str(exp);
      else {
        meta["no-inspect"] = true;
        final = safeStringify(exp);
      }

File: deploy/core/main.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 "use strict";
 
-var app = require('app'),  // Module to control application life.
-    BrowserWindow = require('browser-window'),  // Module to create native browser window.
-    ipc = require("ipc"),
+var app = require('electron').app,  // Module to control application life.
```
[the remark this answers is not in the document]
Comment (Mouvedia) [reply deploy/core/main.js]: https://www.jslint.com/help.html#node

File: deploy/core/main.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+    BrowserWindow = require('electron').BrowserWindow,  // Module to create native browser window.
```
[the remark this answers is not in the document]
Comment (Mouvedia) [reply deploy/core/main.js]: https://www.jslint.com/help.html#node

File: deploy/core/main.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
+    ipcMain = require("electron").ipcMain,
```
[the remark this answers is not in the document]
Comment (Mouvedia) [reply deploy/core/main.js]: https://www.jslint.com/help.html#node

File: deploy/core/main.js [hunk 5]
```diff
@@ -49,7 +49,7 @@ function createWindow() {
   });
 
   // and load the index.html of the app.
-  window.loadUrl('file://' + __dirname + '/LightTable.html?id=' + window.id);
+  window.loadURL('file://' + __dirname + '/LightTable.html?id=' + window.id);
```
[the remark this answers is not in the document]
Comment (Mouvedia) [reply deploy/core/main.js]: https://www.jslint.com/help.html#node

**Comment by Mouvedia** [comment]
Are all the [sub-tasks](https://github.com/LightTable/LightTable/issues/1973#issuecomment-279266658) resolved?

**Comment by prertik** [comment]
Of course not. The target of merging this to `develop` branch was so that, I can further build on the base work of @sbauer322's patches. And, this `develop` branch will be kept dirty for all the works. Previously, I explained it to you the breakage correlation of updating electron and updating Clojurescript. Now, I have some plans regarding this which my future commits will demonstrate.

**Pull request merged**